src/content/docs/en/basics/rendering.mdx

```diff
@@ -0,0 +1,71 @@
+---
+title: "Rendering"
+i18nReady: true
+---
+import PackageManagerTabs from '~/components/tabs/PackageManagerTabs.astro';
+import RecipeLinks from '~/components/RecipeLinks.astro';
+import ReadMore from '~/components/ReadMore.astro';
+
+Your Astro project code must be **rendered** to HTML in order to be displayed on the web. 
+
+Astro pages, routes, and API endpoints can be either [pre-rendered at build time](#prerendered-pages) as static pages or [rendered on demand by a server](#on-demand-rendered-routes) when a route is requested. On-demand rendering requires an adapter to execute code on the server, while prerendered pages are served as static HTML.
+
+- `output: 'static'` (default): Routes are prerendered at build time. You can opt-out of prerendering on any individual route.
+- `output: 'server'` (app mode): All routes are rendered on the server on demand by default. You can opt-in to prerendering on any individual route.
+
+<ReadMore>See more about the [`output` setting](/en/reference/configuration-reference/#output) in the configuration reference.</ReadMore>
+
+**By default, your Astro entire astro site will be prerendered** and static HTML pages will be sent to the browser. If you have some pages that require server-rendering, for example a page that checks for cookies and displays personalized content, you can choose to render these individual routes on demand. The rest of your site will be a static site.
+
+For a **highly dynamic app**, set your build output configuration to server-render all your pages by default. Then, you can choose to prerender any individual pages that do not require a server to execute, such as a privacy policy or about page.
+
+## Opting in or out of prerendering
+
+Both of Astro's rendering modes allow you to prerender and to render on demand in the same project. The only difference is which kind of rendering happens by default.
+
+You can set a `prerender` value (`true` or `false`) in your page route to opt in or out of prerendering and override your output's default mode.
+
+<ReadMore>See more about [adding an adapter and using on-demand rendering](/en/guides/on-demand-rendering/) for some or all of your routes.</ReadMore>
+
+### Prerendered pages
+
+**`'static'`:** by default
+
+**`'server'`:** with `export const prerender = true`
```

Comment:
nit: layout suggestion
I wonder if we shouldn't take the same approach as `Types:`, `Default:`... to avoid the spacing between the two:
```suggestion
<p>

**`'static'`:** by default<br/>
**`'server'`:** with `export const prerender = true`
</p>
```
In my mind the two are related, the spacing is not justified but as I said, it's nitpicking, not really important.

Reply:
Yup, I like it! Consistency!